mesonbuild/interpreter/interpreter.py

```diff
@@ -1297,15 +1298,17 @@ def func_project(self, node: mparser.FunctionNode, args: T.Tuple[str, T.List[str
 
         self.build.projects[self.subproject] = proj_name
         mlog.log('Project name:', mlog.bold(proj_name))
-        mlog.log('Project version:', mlog.bold(self.project_version))
+        mlog.log('Project version:', mlog.bold(self.project_version or "unset"))
 
         if not self.is_subproject():
             # We have to activate VS before adding languages and before calling
             # self.set_backend() otherwise it wouldn't be able to detect which
             # vs backend version we need. But after setting default_options in case
             # the project sets vs backend by default.
             backend = self.coredata.get_option(OptionKey('backend'))
+            assert backend is None or backend is str
             vsenv = self.coredata.get_option(OptionKey('vsenv'))
+            assert vsenv is bool
```

Comment:
These asserts are odd. You're asserting that they are objects of type `Type[bool]` etc rather than that they are instances of said type, which I think is sure to fail.

I believe we usually cast this though.

Reply:
Hah, you reviewed this one minute before I fixed it :)

What does casting look like?

Reply:
I have been convinced to not cast these but to assert because we have changed/added to the type in the past, and the assert will catch that